.typos.toml

```diff
@@ -0,0 +1,28 @@
+# Configuration for https://github.com/crate-ci/typos
+
+[files]
+extend-exclude = [
+  # Third party.
+  "build-aux/freestanding-toolchain",
+  "freestanding-headers",
+  "decompressor/cc-runtime",
+  "limine-efi",
+  "tinf",
+  "common/flanterm",
+  "stb",
+]
```

Comment:
Is this necessary at all considering the workflow doesn't run `./bootstrap`?

Reply:
Yes. Imagine you run `$ typos` locally, after one ran `./bootstrap` beforehand.

Reply:
Well, in that case shouldn't this include all files/directories listed by `git status --ignored` as being ignored after running `./bootstrap` (and potentially after `./configure --enable-all && make` as well)?

Reply:
I updated the PR. We can remove the excludes as the are coming from `.gitignore` and typos automatically uses common ignore files